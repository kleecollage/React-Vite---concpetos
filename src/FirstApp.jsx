import { PropTypes } from "prop-types"; // se instala con yarn add prop-types o si se usa npm npm intall prop-types

const newMessage = 'Soy la constante newMessage';
const num = 455;
const num2 = 333;
const bool = true;
const arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
const obj = {
    message: 'Hola Mundo',
    title: 'Soy un objeto'
}

const suma = () => {
    let resultado = num + num2;
    return resultado
}

// export default function FirstApp(props) { // Lo comun es usar desestructuracion....
export default function FirstApp({
    title = 'title: Soy un default value property',
    subTitle = 'subtitle: soy un default prop',
    propValue = 5,
    name}) {
    // if (!propValue) {
    //     throw new Error('porpValue no existe')
    // }
    // console.log(props) // props = properties
    return (
        // el atajo <> es l aabreviatura de fragment y nos evita el tener que importarlo
        <> 
            {/* <h1>{title}</h1>
            <h2>{subTitle + 1}</h2>
            <h1>FirstApp</h1>
            <h2>by Antonio Hernandez Santander</h2>
            <h3>{3 * 2}</h3>
            <h4>{newMessage}</h4>
            <h5>{num}</h5>
            <h6>{bool}</h6>
            <p>{arr}</p>
            <code>{JSON.stringify (obj)}</code>
            <div>{ suma() }</div> */}
            <h1 data-testid='test-title'>{title}</h1>
            <div>{subTitle}</div>
            <div>{subTitle}</div>
            <h3>{name}</h3>
            <p>{propValue}</p>
        </>
  )
}

FirstApp.propTypes = {
    propValue: PropTypes.number.isRequired,
    subTitle: PropTypes.string.isRequired 
}

// FirstApp.defaultProps = {
//   subTitle: "subTitle (default): valor definido en defaultProps",
//   propValue: 0,
//   name: "name (default): Antonio Hernandez",
// };

