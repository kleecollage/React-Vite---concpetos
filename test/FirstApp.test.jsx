import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";

describe("Prueba en FirstApp", () => {
    // test("debe hacer match con el snapshot", () => {
    //     const propValue = 5;
    //     const subTitle = 'subTitle 2'
    //     const { container } = render(
    //         <FirstApp propValue={propValue} subTitle={subTitle} />
    //     );
    //     expect(container).toMatchSnapshot()
    // });
    
    test('debe de mostrar el titulo en un h1', () => {
        const title = "title: Soy un default value property";
        const subTitle = "subtitle: soy un default prop";
        const propValue = 5;
        const { container, getByText, getByTestId } = render(
            <FirstApp propValue={propValue} subTitle={subTitle} />
        );
        expect(getByText(title)).toBeTruthy();
        
        const h1 = container.querySelector('h1');
        console.log(h1.innerHTML)
        expect(h1.innerHTML).toContain(title)
        expect(getByTestId('test-title').innerHTML).toContain(title)
    })

    test('debe de mostrar el subtitulo enviado por props', () => { 
        const title = "title: Soy un default value property";
        const subTitle = "subTitle 2";
        const propValue = 5;
        const { getAllByText } = render(
            <FirstApp
                title={title}
                propValue={propValue}
                subTitle={subTitle}
            />
        );
        expect(getAllByText(subTitle).length).toBe(2);
     })
});
