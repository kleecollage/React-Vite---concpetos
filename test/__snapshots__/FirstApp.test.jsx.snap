// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Prueba en FirstApp debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    title: Soy un default value property
  </h1>
  <h2>
    subTitle 2
  </h2>
  <h3 />
  <p>
    5
  </p>
</div>
`;
